src/audio/codec_adapter/codec/cadence.c

```diff
@@ -571,6 +571,8 @@ static int cadence_codec_reset(struct comp_dev *dev)
 			ret);
 	}
 
+	codec->mpd.init_done = 0;
+
 	return ret;
```

Comment:
Please move codec->mpd.init_done before cadence_codec_prepare. In case of of DAP codec cadence_codec_prepare() updates this field so you are resetting int now.